plugins/chat/assets/javascripts/discourse/services/chat-channel-thread-pane-subscriptions-manager.js

```diff
@@ -0,0 +1,39 @@
+import ChatChannelPaneSubscriptionsManager from "./chat-channel-pane-subscriptions-manager";
+import ChatMessage from "discourse/plugins/chat/discourse/models/chat-message";
+import { handleStagedMessage } from "discourse/plugins/chat/discourse/lib/staged-message-handler";
+
+export default class ChatChannelThreadPaneSubscriptionsManager extends ChatChannelPaneSubscriptionsManager {
+  get messageBusChannel() {
+    return `/chat/${this.model.channelId}/thread/${this.model.id}`;
+  }
+
+  get messageBusLastId() {
+    return this.model.threadMessageBusLastId;
+  }
+
+  // NOTE: This is a noop, there is nothing to do when a thread is created
+  // inside the thread panel.
+  handleThreadCreated() {
+    return;
+  }
+
+  // TODO (martin) Hook this up in Chat::Publisher, it doesn't work yet there
+  // because we only have the deleted message IDs, not the message model.
+  handleBulkDeleteMessage() {
+    return;
+  }
```

Comment:
Maybe we create a base service with all the logic and have two children:
- one for channel
- one for thread

Which implement only what is needed

Reply:
We could have the `handleStagedMessage` code in this base service instead

Reply:
Yep that's fair enough I can do that

Reply:
Actually thinking about this a bit more, 90% of the handling code is the same between thread and channel. I guess the main benefit of having this all in a base class is so the channel can do different things as needed too?